### PR TITLE
Change project.json projects' cache file path

### PR DESCRIPTION
**Pull request by nkolev92**
Change the project.json projects' cache file path to be in the base intermediate output path.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRequest.cs [hunk 1]
```diff
@@ -136,6 +136,11 @@ public class RestoreRequest
         public string RestoreOutputPath { get; set; }
 
         /// <summary>
+        /// </summary>
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRequest.cs]: nit: add a summary or remove the tags

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs [hunk 2]
```diff
@@ -29,27 +29,16 @@ internal static bool IsNoOpSupported(RestoreRequest request)
...
-            {
-                var projFileName = Path.GetFileName(request.Project.RestoreMetadata.ProjectPath);
-                return request.Project.RestoreMetadata.CacheFilePath = Path.Combine(request.RestoreOutputPath, $"{projFileName}.nuget.cache");
-            }
-
-            return null;
-        }
-
-        internal static string GetProjectJsonCacheFilePath(RestoreRequest request)
+        internal static string GetPJorPRCacheFilePath(RestoreRequest request)
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs:32]: GetCacheFilePath?

File: src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs [hunk 3]
```diff
@@ -208,6 +208,7 @@ private static ProjectRestoreMetadata GetMSBuildMetadata(PackageSpec packageSpec
             msbuildMetadata.ProjectJsonPath = rawMSBuildMetadata.GetValue<string>("projectJsonPath");
             msbuildMetadata.ProjectName = rawMSBuildMetadata.GetValue<string>("projectName");
             msbuildMetadata.ProjectPath = rawMSBuildMetadata.GetValue<string>("projectPath");
+            msbuildMetadata.OutputPath = rawMSBuildMetadata.GetValue<string>("outputPath");
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs]: was this broken before?

**Review by emgarten — APPROVED** [review]
LGTM 🚀

File: src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs (hunk 3 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs]: No, I added it twice. 
Removed it now.

**Pull request merged**